src/test-project/src/test/java/com/redhat/devtools/intellij/commonUiTestLibrary/fixturesTest/dialogs/projectManipulation/NewProjectDialogTest.java

```diff
@@ -11,145 +11,389 @@
+        mavenGradleFinalPage.openArtifactCoordinates();
+        String currentArtifactId = mavenGradleFinalPage.getArtifactId();
+        String newArtifactId = currentArtifactId + "1";
+        mavenGradleFinalPage.setArtifactId(newArtifactId);
+        currentArtifactId = mavenGradleFinalPage.getArtifactId();
+        assertTrue(currentArtifactId.equals(newArtifactId), "Currently set artifact ID should be '" + newArtifactId + "' but is '" + currentArtifactId + "'.");
     }
 
-    private void testProjectNameInputField(String projectName, NewProjectType newProjectType) {
+    private void testGetSetVersionMavenGradle(NewProjectType newProjectType) {
         navigateToSetProjectNamePage(newProjectType);
-        NewProjectDialog newProjectDialog = remoteRobot.find(NewProjectDialog.class, Duration.ofSeconds(10));
-        newProjectDialog.setProjectName(projectName);
-        String projectNameFromInputField = remoteRobot.findAll(JTextFieldFixture.class, JTextFieldFixture.Companion.byType()).get(0).getText();
-        assertTrue(projectName.equals(projectNameFromInputField), "Project name in the input field (" + projectNameFromInputField + ") is different from the expected project name (" + projectName + ").");
+        MavenGradleNewProjectFinalPage mavenGradleFinalPage = newProjectDialogWizard.find(MavenGradleNewProjectFinalPage.class, Duration.ofSeconds(10));
+        mavenGradleFinalPage.openArtifactCoordinates();
+        String currentVersion = mavenGradleFinalPage.getVersion();
+        String newVersion = currentVersion + "1";
+        mavenGradleFinalPage.setVersion(newVersion);
+        currentVersion = mavenGradleFinalPage.getVersion();
+        assertTrue(currentVersion.equals(newVersion), "Currently set version should be '" + newVersion + "' but is '" + currentVersion + "'.");
+    }
+
+    private void makeSureMoreSettingsIsClosed(JavaNewProjectFinalPage javaFinalPage) {
+        if (isMoreSettingsOpened(javaFinalPage)) {
+            javaFinalPage.jLabel(ButtonLabels.moreSettings).click();
+        }
+    }
+
+    private boolean isMoreSettingsOpened(JavaNewProjectFinalPage javaFinalPage) {
+        return javaFinalPage.findAll(ContainerFixture.class, byXpath("//div[@class='TitledSeparator']/../../*")).size() == 2;
+    }
+
+    private void makeSureArtifactCoordinatesIsClosed(MavenGradleNewProjectFinalPage mavenGradleFinalPage) {
+        if (isArtifactCoordinatesOpened(mavenGradleFinalPage)) {
+            mavenGradleFinalPage.jLabel(ButtonLabels.artifactCoordinates).click();
+        }
+    }
+
+    private boolean isArtifactCoordinatesOpened(MavenGradleNewProjectFinalPage mavenGradleFinalPage) {
+        List<ContainerFixture> cf = mavenGradleFinalPage.findAll(ContainerFixture.class, byXpath("//div[@class='HideableTitledSeparator']/../*"));
+        return cf.size() > 5;
     }
 
     private enum NewProjectType {
```

Comment:
Isn't it a shame not to use this enum on multiple places? Becuase I believe I have seen other place you could make use of it.